runtime/capsules/src/mctp/driver.rs

```diff
@@ -0,0 +1,355 @@
+
+                        let res = kernel_data
+                            .get_readonly_processbuffer(ro_allow::MESSAGE_WRITE)
+                            .and_then(|write| {
+                                write.enter(|wpayload| {
+                                    self.kernel_msg_buf.take().map_or(
+                                        Err(ErrorCode::NOMEM),
+                                        |mut kernel_msg_buf| {
+                                            if wpayload.len() > kernel_msg_buf.len() {
+                                                return Err(ErrorCode::SIZE);
+                                            }
+                                            wpayload.copy_to_slice(
+                                                &mut kernel_msg_buf[..wpayload.len()],
+                                            );
+                                            kernel_msg_buf.slice(0..wpayload.len());
+
+                                            match self.sender.send_msg(
+                                                msg_type,
+                                                dest_eid,
+                                                msg_tag,
+                                                kernel_msg_buf,
+                                            ) {
+                                                Ok(_) => {
+                                                    println!("MCTPDriver: send_msg success");
+                                                    app.pending_tx = Some(OpContext {
+                                                        msg_tag,
+                                                        peer_eid: dest_eid,
+                                                        msg_type,
+                                                        op_type: OpType::Tx,
+                                                    });
+                                                    self.current_app.set(Some(process_id));
+                                                    Ok(())
+                                                }
+                                                Err(mut buf) => {
+                                                    println!("MCTPDriver: send_msg failed");
+                                                    buf.reset();
+                                                    self.kernel_msg_buf.replace(buf);
+                                                    Err(ErrorCode::FAIL)
+                                                }
+                                            }
+                                        },
+                                    )
+                                })
+                            })
+                            .unwrap_or(Err(ErrorCode::FAIL));
+                        match res {
+                            Ok(()) => Ok(()),
+                            Err(e) => Err(e),
+                        }
+                        // Ok(())
```

Comment:
```suggestion
                        res
```
?

Reply:
Refactored this part of the code. Let me know if this looks okay now.